src/core/baseClient.ts

```diff
@@ -302,4 +305,49 @@ export class BaseClient {
       )
     );
   }
+  /**
+   * Gets the current student's rewards shop
+   * @returns Array of purchasable items
+   */
+  async getRewards(): Promise<RewardsResponse> {
+    return (
+      await this.makeAuthedRequest(
+        this.API_BASE + "/rewards/" + this.studentId,
+        {
+          method: "GET",
+        },
+      )
+    );
+  }
+  /**
+   * Purchase a reward item from the current student's  rewards shop
+   * @param itemId number
+   * @returns An object containg balence and id
+   */
+  async postRewardPurchase(itemId: number): Promise<RewardPurchaseResponse> {
+    return (
+      await this.makeAuthedRequest(
+        this.API_BASE + "/purchase" + itemId,
+        {
+          method: "POST",
+          body: `pupil_id=${this.studentId}`
+        }
+      )
+    )
+  }
+
+  /**
+   * Gets the current student's pupil fields
+   * @returns Array of stats
+   */
+  async getPupilFields(): Promise<PupilFieldsResponse> {
+    return (
+      await this.makeAuthedRequest(
+        this.API_BASE + "/apiv2student/customfields" + this.studentId,
```

Comment:
```suggestion
        this.API_BASE + "/apiv2student/customfields/" + this.studentId,
```
Missing forward slash